include/boost/multiprecision/gmp.hpp

```diff
@@ -300,10 +300,8 @@ struct gmp_float_imp
          return *this;
       }
 
-      #ifndef BOOST_MP_STANDALONE
-      BOOST_MP_ASSERT(!(boost::math::isinf)(a));
-      BOOST_MP_ASSERT(!(boost::math::isnan)(a));
-      #endif
+      BOOST_MP_ASSERT(!(boost::multiprecision::detail::isinf)(a));
```

Comment:
This code has to work with float, double, long double and __float128, I think we can use `isinf` etc as unqualified calls (with macro expansion suppression), and update the using declarations at the start, plus BOOST_MP_FLOAT128_USING to bring the functions into scope.